### PR TITLE
[AIRFLOW-1561] Fix scheduler to pick up example DAGs without other DAGs

### PR DESCRIPTION
**Pull request by mrkm4ntr**
Dear Airflow maintainers,

Please accept this PR. I understand that it will not be reviewed until I have checked off all the steps below!

### JIRA
- [x] My PR addresses the following [Airflow JIRA](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "[AIRFLOW-XXX] My Airflow PR"
    - https://issues.apache.org/jira/browse/AIRFLOW-1561

### Description
- [x] Here are some details about my PR, including screenshots of any UI changes:
The scheduler doesn't pick up example dags unless there is at least 1 dag in dags folder. This is confusing for those who try it for the first time.

### Tests
- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:
* tests.jobs.SchedulerJobTest.test_list_py_file_paths

### Commits
- [x] My commits all reference JIRA issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
    1. Subject is separated from body by a blank line
    2. Subject is limited to 50 characters
    3. Subject does not end with a period
    4. Subject uses the imperative mood ("add", not "adding")
    5. Body wraps at 72 characters
    6. Body explains "what" and "why", not "how"

### CODE REVIEW

File: airflow/utils/dag_processing.py [hunk 1]
```diff
@@ -207,6 +209,11 @@ def list_py_file_paths(directory, safe_mode=True):
                 except Exception:
                     log = LoggingMixin().log
                     log.exception("Error while examining %s", f)
+    if include_examples:
+        example_dag_folder = os.path.join(
+            os.path.dirname(os.path.dirname(__file__)),
+            'example_dags')
```
Comment (edgarRd) [inline airflow/utils/dag_processing.py]: Would be good to have literal `'example_dags'` as a common constant.

**Comment by xnuinside** [comment]
@mrkm4ntr, will you update PR and squash commits? 
@kaxil, @ashb, can you review this PR? It was relative to several tasks, including https://issues.apache.org/jira/projects/AIRFLOW/issues/AIRFLOW-3118 

if the author will not answer, maybe make sense just reopen it as PR for https://issues.apache.org/jira/projects/AIRFLOW/issues/AIRFLOW-3118

**Comment by mrkm4ntr** [comment]
@xnuinside Sure. However I would like to start after the reviewers are assigned.

**Comment by kaxil** [comment]
Looks logical, can you rebase the master please. Once that is done I will test this and if everything works file, will merge 👍

**Comment by mrkm4ntr** [comment]
@kaxil I rebased. Please check it.

**Review by kaxil — APPROVED** [review]
LGTM

**Comment by kaxil** [comment]
@ashb @Fokko Any comments, if none - I'll merge this. I tested this PR and it works fine and fixes the issue.

A minor issue I have is "it says 16-17 dags in DAGBAG" when `load_examples = True` and there are no dags in DAGS folder. It says that because of the example DAGs but might be confusing for users. However, I don't have a major issue with that.

**Review by Fokko — APPROVED** [review]
LGTM, much cleaner

**Comment by XD-DENG** [comment]
Hi @Fokko @kaxil , seems the Travis CI testing time needed for the two `backend-sqlite` tests increased from ~30 minutes to ~40 minutes since this PR is merged.

May worth having a check.

**Comment by ashb** [comment]
We probably need to not disable loading all DAGs and instead only specific dags in many tests.

**Comment by XD-DENG** [comment]
The weird thing is only the two `backedn-sqlite` tests are affected. The time needed by `mysql`/`postgres` tests remain roughly the same.

**Comment by kaxil** [comment]
Good find @XD-DENG , will need to look into it and as Ash  mentioned we might want to stop loading all dags

**Pull request merged**